modules/packages/UnrolledLinkedList.chpl

```diff
@@ -0,0 +1,1249 @@
+      //TODO: Maybe some optimization here. This O(N^2)
+      var result = false;
+
+      _sanity(size >= 0);
+
+      if size == 0 then
+        return true;
+
+
+      on this {
+        if idx == _size {
+          for x in items {
+            _append(x);
+          }
+          result = true;
+        }
+        else if _withinBounds(idx) {
+          var count = 0;
+          for x in items {
+            _insert(idx + count, x);
+            count += 1;
+          }
+          result = true;
+        }
+      }
+
+      return result;
+    }
+
+    /*
+      Insert an array of elements `arr` into this unrolledLinkedList at index `idx`,
+      shifting all elements at and following the index `arr.size` positions
+      to the right. 
+
+      If the insertion is successful, this method returns `true`. If the given
+      index is out of bounds, this method does nothing and returns `false`.
+
+      .. warning::
+
+        Inserting elements into this unrolledLinkedList may invalidate existing references
+        to the elements contained in this unrolledLinkedList.
+
+      :arg idx: The index into this unrolledLinkedList at which to insert.
+      :type idx: `int`
+
+      :arg arr: An array of elements to insert.
+      :type x: `[] eltType`
+
+      :return: `true` if `arr` was inserted, `false` otherwise.
+      :rtype: `bool`
+    */
+    proc ref insert(idx: int, arr: [?d] eltType): bool lifetime this < arr {
+
+      var result = false;
+
+      on this {
+        _enter();
+        result = _insertGenericKnownSize(idx, arr, arr.size);
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Insert a list of elements `lst` into this unrolledLinkedList at index `idx`, shifting
+      all elements at and following the index `lst.size` positions to the
+      right.
+
+      If the insertion is successful, this method returns `true`. If the given
+      index is out of bounds, this method does nothing and returns `false`.
+
+      .. warning::
+
+        Inserting elements into this unrolledLinkedList may invalidate existing references
+        to the elements contained in this unrolledLinkedList.
+
+      :arg idx: The index into this unrolledLinkedList at which to insert.
+      :type idx: `int`
+
+      :arg lst: A list of elements to insert.
+      :type lst: `list(eltType)`
+
+      :return: `true` if `lst` was inserted, `false` otherwise.
+      :rtype: `bool`
+    */
+    proc ref insert(idx: int, lst: list(eltType)): bool lifetime this < lst {
+      
+      var result = false;
+      
+      // Prevent deadlock if we are trying to insert this into itself.
+      const size = lst.size;
+
+      on this {
+        _enter();
+        result = _insertGenericKnownSize(idx, lst, size);
+        _leave();
+      }
+
+      return result;
+    }
+
+    /*
+      Remove the first `count` elements from this unrolledLinkedList with values equal to
+      `x`, shifting all elements following the removed item left.
```

Comment:
If it was me, I would turn this sentence around. The purpose of this function is not to remove `count` elements. It is to remove an element. Optionally, you can pass a `count` argument to repeat this.